vnext/PropertySheets/ReactDirectories.props

```diff
@@ -15,14 +15,17 @@
 
     <ReactNativePackageDir Condition="'$(ReactNativePackageDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), 'node_modules\react-native\package.json'))\node_modules\react-native\</ReactNativePackageDir>
 
+    <RootDir>$(ReactNativeWindowsDir)</RootDir>
```

Comment:
we should limit the amount of special sauce needed. Can we just reuse the standard OutDir and IntDir instead?

Reply:
This gives flexibility for custom builds.
MSVC doesn't really set up an output artifact structure that consolidates `Platform`, `Configuration`, and project path.
These additions allow for placing either the intermediate outputs, final outputs, or both in a different directory with very compact expressivity.

In short, they're just transparent build variables that won't affect the average case.